### ISSUE TITLE
Explain dependency on hashbrown

### ISSUE THREAD

**Issue by pitaj** [udoprog/fixed-map#6]
Wanted to request a little more documentation on this point. 

Based on the array explanation, it's not clear why a second hash map is required. For instance, if you're just using a C style enum as keys, does the hash brown map come into play? And what are the performance implications when it does come into play?

Thanks

**Comment by udoprog** [comment]
Hey!

`hashbrown` is used when the key is not statically known, as would be the case with the `String` or `u32` variants here:

```rust
#[derive(Clone, Copy, Key)]
enum Key {
    Simple,
    String(&'static str),
    Number(u32),
    Singleton(()),
}
```

In #8 I'm making the dependency to `hashbrown` optional through the `map` feature, I hope this helps!

**Comment by udoprog** [comment]
Note that it won't be released until Rust 1.65 since that's when GATs are stable as well!

**Comment by pitaj** [comment]
Hey that's wonderful news. Thank you, I eagerly await the stabilization of GATs 🎊
